### PR TITLE
chore(sample): Remove old sample in favour of the new arch sample

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Refactoring

## :scroll: Description

The original sample is not being updated and the update rn ci script stopped working. Instead of trying to fix it, I've removed the old sample and fixed ci to use the new sample in places where it was needed.

The update-rn.sh originally updated the rn version in the sample. I've changed it to update the dev dep RN version of the SDK, so our SDK is always built with the latest RN.

`update-rn.sh` will be updated in the future to actually update the RN version in all the places thru the repo. But for now, let's limit it only to the SDK `package.json`.

## :green_heart: How did you test it?

CI

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [ ] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps
#skip-changelog

### CODE REVIEW

**Comment by lucas-zimerman** [comment]
Nice one!

A few points:

I see you removed the old sample but it seems like some references about it weren't removed, some of them I can't request a change since it's none of the files were altered, but here are the files:

.eslintignore
https://github.com/getsentry/sentry-react-native/blob/main/.eslintignore#L5

.vscode
https://github.com/getsentry/sentry-react-native/blob/main/.vscode/settings.json#L26

Other than those minor changes, it looks ready for merge, thank you!

**Review by lucas-zimerman — APPROVED** [review]

File: .github/workflows/sample-application.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            build \
+            | tee xcodebuild.log \
+            | xcbeautify --quieter --is-ci --disable-colored-output
+
+      - name: Upload logs
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-sample-${{ matrix.rn-architecture }}-${{ matrix.platform }}-${{ matrix.build-type }}-logs
+          path: sample-new-architecture/ios/*.log
```
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply .github/workflows/sample-application.yml:119]: The Android build logs are directly in the console but iOS logs are super long so we save them to text file.

**Comment by krystofwoldrich** [comment]
@lucas-zimerman Thank you, I've missed those references.

**Comment by marandaneto** [comment]
@krystofwoldrich if we don't build the old sample, how do we know that the SDK is compatible with older versions and new architecture disabled?

**Comment by krystofwoldrich** [comment]
@marandaneto We have separate ci jobs that create plain RN app of given RN version, add Sentry, build the app with given arch (new/legacy) and run e2e on that app.

https://github.com/getsentry/sentry-react-native/actions/runs/5315403231/jobs/9623722682?pr=3118

https://github.com/getsentry/sentry-react-native/blob/e727df81268641c0729b2cdd24dec0b1c3832ce2/.github/workflows/e2e.yml#L244


The new sample is built both for legacy and new arch.
https://github.com/getsentry/sentry-react-native/actions/runs/5315403229/jobs/9623722294?pr=3118

**Pull request merged**